terraform/modules/shared/main.tf

```diff
@@ -472,8 +472,49 @@ resource "azurerm_kubernetes_cluster" "cluster" {
   }
 
   identity {
-    type         = "UserAssigned"
-    identity_ids = [azurerm_user_assigned_identity.pipeline_runner.id]
+    type = "SystemAssigned"
+  }
+
+}
+
+data "azurerm_kubernetes_cluster" "credentials" {
+  name                = azurerm_kubernetes_cluster.cluster.name
+  resource_group_name = var.resource_group_name
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.azurerm_kubernetes_cluster.credentials.kube_config.0.host
+    client_certificate     = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.client_certificate)
+    client_key             = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.client_key)
+    cluster_ca_certificate = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.cluster_ca_certificate)
+  }
+}
+
+resource "helm_release" "record_linkage" {
+  name          = "phdi-${terraform.workspace}"
+  repository    = "https://cdcgov.github.io/phdi-charts/"
```

Comment:
Quick note on these three lines:

* The repository is where the charts are actually hosted
* `chart` is the name of the specific chart we want to install - so it's not mutable
* `recreate_pods` forces the pods to restart on a deployment. Turns out that when Secret objects are updated in Kubernetes, the pods that rely on them _don't_ automatically update. Much debugging later, you can force an update with this flag, which will tear down and recreate the pods. (But thanks to K8s magic, there should be no service interruptions)